public/js/components/SourcesTree.js

```diff
@@ -158,7 +158,7 @@ let SourcesTree = React.createClass({
         return [];
       },
       getRoots: () => sourceTree.contents,
-      getKey: (item, i) => item.path,
+      getKey: (item, i) => item.path + i,
```

Comment:
You don't want to do this because it destroys the purpose of the key. The keys should stay consistent for each item with the renders, so that React can easily figure out which ones have simply moved. This makes React think items have been removed and need to be re-added.

The point of `item.path` is to always be unique. When is that not unique? That's a bug, and we should fix the path generation if there is a bug in there.